lib/msf/core/modules/metadata/search.rb

```diff
@@ -192,19 +211,30 @@ def is_match(params, module_metadata)
           break if match
         end
         # Filter this module if no matches for a given keyword type
-        if mode == 0 and not match
+        if mode == SearchMode::INCLUDE and not match
           return false
         end
       end
       # Filter this module if we matched an exclusion keyword (-value)
-      if mode == 1 and match
+      if mode == SearchMode::EXCLUDE and match
         return false
       end
     end
 
     true
   end
 
+  def as_regex(search_term)
+    # Convert into a case-insensitive regex
+    utf8_buf = search_term.dup.force_encoding('UTF-8')
+    if utf8_buf.valid_encoding?
+       Regexp.new(Regexp.escape(utf8_buf), Regexp::IGNORECASE)
+    else
+      # If the encoding is invalid, default to a regex that matches anything
+      //
```

Comment:
This defaulting behaior was requested to be implemented as part of https://github.com/rapid7/metasploit-framework/pull/13282 such that if there's an encoding issue - it's silently hidden from the user. Following the zen of python I'm tempted to raise an exception instead